utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/sink/KafkaAvroSinkFactory.scala

```diff
@@ -0,0 +1,22 @@
+package pl.touk.nussknacker.engine.avro.sink
+
+import pl.touk.nussknacker.engine.api.LazyParameter
+import pl.touk.nussknacker.engine.api.process.{Sink, SinkFactory}
+import pl.touk.nussknacker.engine.avro.RuntimeSchemaData
+import pl.touk.nussknacker.engine.avro.encode.ValidationMode
+import pl.touk.nussknacker.engine.kafka.serialization.KafkaSerializationSchema
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, PreparedKafkaTopic}
+import pl.touk.nussknacker.engine.util.KeyedValue
+
+trait KafkaAvroSinkFactory extends SinkFactory {
```

Comment:
I wonder, do we need inheritance here? It seems that everything is passed in arguments?

Reply:
I don't understand. `KafkaAvroSinkFactory extends SinkFactory` just to not add `extends SinkFactory` in all concrete implementations. How u would like to it in other way?

Reply:
Replaced by composition - like in sources case